pyproject.toml

```diff
@@ -31,25 +28,47 @@ requires-python = ">=3.10"
 # Make sure to list one dependency per line.
 dependencies = [
   "dask",
-  "graphviz",
   "essreduce",
+  "graphviz",
   "plopp",
   "pythreejs",
   "sciline>=24.4.1",
-  "scipp>=23.8.0",
-  "scippneutron>=23.9.0",
-  "scippnexus>=23.12.1",  # 23.12.0 and below deadlock in threaded use
+  "scipp>=23.8",
+  "scippneutron>=23.9",
+  "scippnexus>=23.12.1", # 23.12.0 and below deadlock in threaded use
 ]
 
-dynamic = ["version"]
+dynamic = [
+  "version",
+]
 
 [project.urls]
 "Bug Tracker" = "https://github.com/scipp/esssans/issues"
 "Documentation" = "https://scipp.github.io/esssans"
 "Source" = "https://github.com/scipp/esssans"
-
 [tool.setuptools_scm]
 
+[tool.ruff.lint]
+select = ["B", "C4", "DTZ", "E",  "F", "FA",
+          "G", "I", "S", "T20", "PERF", "PGH",
+          "PT", "PYI", "RUF", "UP", "W"]
+# FIXME: This shouldn't be required but tests
+# fail when PEP 585 is followed
+ignore = ['UP006', 'UP035', 'UP007']
+pydocstyle.convention = "numpy"
```

Comment:
This needs to be fixed.

Reply:
What are you referring to? The upgrade lints? They could be done right now.

The docstyle setting should do nothing right now because `D`-rules are not enabled.

Reply:
> What are you referring to? The upgrade lints? They could be done right now.

It fails the test for some reason! I will dig into this a bit later.